zigup.zig

```diff
@@ -636,3 +683,16 @@ pub fn appendCommandString(appender: *appendlib.Appender(u8), argv: []const []co
         prefix = " ";
     }
 }
+pub fn shouldDeleteCompiler(master_points_to_opt: ?[]const u8, default_compiler_opt: ?[]const u8, name: []const u8) enum { yes, no_is_default, no_is_master } {
+    if (default_compiler_opt) |default_comp| {
+        if (mem.eql(u8, default_comp, name)) {
+            return .no_is_default;
+        }
+    }
+    if (master_points_to_opt) |master_points_to| {
+        if (mem.eql(u8, master_points_to, name)) {
+            return .no_is_master;
+        }
+    }
+    return .yes;
+}
```

Comment:
Here's an idea to simplify a bit.  We can have this return an optional string like this:
```zig
pub fn getKeepReason(master_points_to_opt: ?[]const u8, default_compiler_opt: ?[]const u8, name: []const u8) ?[]const u8 {
    if (default_compiler_opt) |default_comp| {
        if (mem.eql(u8, default_comp, name)) {
            return "is default compiler";
        }
    }
    if (master_points_to_opt) |master_points_to| {
        if (mem.eql(u8, master_points_to, name)) {
            return "it is master";
        }
    }
    return null;
}
```

So then instead of caller doing this:
```zig
switch (shouldDeleteCompiler(master_points_to_opt, default_comp_opt, compiler_name)) {
    .yes => {},
    .no_is_default => {
        std.debug.warn("Error: not deleting '{}' (is default compiler)\n", .{compiler_name});
        return;
    },
    .no_is_master => {
        std.debug.warn("Error: not deleting '{}' (because it is master)\n", .{compiler_name});
        return;
    },
}
```

We can do this
```zig
if (getKeepReason(master_points_to_opt, default_comp_opt, compiler_name)) |reason| {
    std.debug.warn("Error: cannot clean '{}' ({})\n", .{compiler_name, reason});
    return error.AlreadyReported;
}
```

Reply:
Nice. I thought of doing the opp where you would pass a comptime format string to shouldDeleteCompiler, but it was too complex.